src/signTx.c

```diff
@@ -1855,3 +1872,52 @@ void signTx_handleAPDU(
 	VALIDATE(subhandler != NULL, ERR_INVALID_REQUEST_PARAMETERS);
 	subhandler(p2, wireDataBuffer, wireDataSize);
 }
+
+ins_sign_tx_aux_data_context_t* accessAuxDataContext()
+{
+	switch (ctx->stage)
+	{
+	case SIGN_STAGE_AUX_DATA:
+	case SIGN_STAGE_AUX_DATA_CATALYST_REGISTRATION_SUBMACHINE:
+		return &(ctx->txPartCtx.aux_data_ctx);
+	
+	default:
+		return NULL;
```

Comment:
I would say we just want to do `ASSERT(false)` here (yes, NULL can be returned to make the compiler happy) and never check the returned value in other places. The value NULL is not meaningful and there is no point in returning it, we must stop processing the tx immediately if a wrong part of the context is accessed.

Reply:
I removed the ASSERT in one of the commits because it doesn't help much when trying to debug. If you call the function directly, you get the assert in the function, and will have no idea where it comes from. Hence why I wrote the macros. If it were not C, the functions would be private and not callable at all from the outside.

Reply:
I've dismissed that on the grounds of having enough tracing to keep one aware of what is being executed (in particular, what stage one is in, which is the determinant of whether the function fails or not). It is true that debugging would be easier if one knew precisely where the context access failed.

But dereferencing a NULL pointer is undefined behavior (not a guaranteed program crash), and I am afraid someone might use the function without the macro (impossible to guard against in C afaik, macros don't provide any scope-controlling features). This is a much bigger concern that longer debugging.

What about having two versions of `accessAuxDataContext` (one for debugging purposes, commented out, the other failing the assertion) and then AUX_DATA_CTX will only define which of them will be called?

We will also save on code size a bit (no redundant NULL checks).

Also, `AUX_DATA_CTX` does not need arguments, it can be treated as a variable when used (although it does involve a function call).

Reply:
The macro structure probably needs to be a bit more complex, but it is essentially doable (maybe you need helper macros to achieve the effect I tried to describe).

Reply:
Just pushed a suggestion, what do you think about that?